lib/primitives/mtx.js

```diff
@@ -1850,6 +1918,51 @@ class CoinSelector {
     return this;
   }
 
+  /**
+   * Fill a tx with inputs
+   * @returns {void}
+   */
+  async selectFill() {
+    // output to send coin values to
+    // naively doing just first output
+    const output = this.tx.outputs[0];
+
+    while(this.index < this.coins.length) {
+      const coin = this.coins[this.index++];
+      if (!this.isSpendable(coin))
+        continue;
+
+      // if the value of the coin is larger than target threshold
+      // and output isn't dust, we can stop filling
+      if (
+        this.threshold &&
```

Comment:
So if we have multiple sorting options, then it would make more sense to skip `coin.value > threshold`, coins.
and check if output.value is dust after we are done filling.

Reply:
Also do we want to have another "minThreshold" or just another check to verify the coins we are spending are not dust ?

Reply:
How would `this.threshold === -1`? It gets initialized to 0 and then someone could pass in something higher if they want a threshold. I think I need to fix the assertion during initialization to assert it's a safe integer and larger than 0.

I don't think it matters if the coins we spend are dust. That's kind of the point of a sweep and fill is to consolidate very small, unspendable UTXOs. I'm not sure minThreshold would be necessary as you could just another `this.selection` option. The upper bound threshold is kind of just to try and not leak too much privacy but unnecessarily sweeping UTXOs that are big enough to keep on their own.

Reply:
I now initialize threshold to 0, so if this is not set with options this will return false.

Reply:
okay. But still you can't assert what is the sort order you can't use `break`. You probably want to skip instead of breaking. Breaking would only make sense if `reverseValue` was used.

Reply:
Ah ok. Yeah, seems obvious but I didn't think about that. What I did instead was just only allow threshold if `reverseValue` is set, since as you point out it doesn't make sense either way. Maybe I'll do this instead.